src/Foreign/JavaScript/Types.hs

```diff
@@ -64,17 +64,49 @@ This is a record type which has the following fields:
     The initial 'CallBufferMode' to use for 'runFunction'.
     It can be changed at any time with 'setCallBufferMode'.
 
+* @jsSSLBind :: Maybe ByteString@
+
+    Bind address for @HTTPS@.
+    @Nothing@ means that the bind address is read from the environment variable @ADDR@.
+    Alternatively, address @0.0.0.0@ is used if this variable is not set.
+
+* @jsSSLCert :: Maybe FilePath@
+
+    Path to SSL certificate file.
+    @Nothing@ means that the path to SSL certificate file is @cert.pem@.
+
+* @jsSSLCert :: Maybe FilePath@
+
+    Path to SSL certificate file.
+    @Nothing@ means that the path to SSL certificate file is @cert.pem@.
+
+* @jsSSLChainCert :: Maybe Bool@
+
+    If it is SSL chain certificate file.
+    @Nothing@ means that False.
+
+* @jsSSLKey :: Maybe FilePath@
+
+    Path to SSL key file.
+    @Nothing@ means that the path to SSL key file is @key.pem@.
+
 (For reasons of forward compatibility, the constructor is not exported.)
 
 -}
 data Config = Config
-    { jsPort       :: Maybe Int           
+    { jsPort       :: Maybe Int
     , jsAddr       :: Maybe ByteString
     , jsCustomHTML :: Maybe FilePath
     , jsStatic     :: Maybe FilePath
     , jsLog        :: ByteString -> IO ()
     , jsWindowReloadOnDisconnect :: Bool
     , jsCallBufferMode :: CallBufferMode
+    , jsForceSSL     :: Bool
```

Comment:
Could you wrap the entire SSL configuration into a `Maybe`, like this?

```suggestion
    , jsUseSSL     :: Maybe ConfigSSL
```

```hs
data ConfigSSL = ConfigSSL
    { jsSSLBind :: ByteString
    , jsSSLCert :: FilePath
    , …
    }
```

The original semantics of `Maybe` for `jsPort` and `jsAddr` was *not* to denote an optional value, but to highlight that the value is taken from the environment. For security reasons, Threepenny-gui probably does not want to take anything from the environment by the default when dealing with SSL. (Library users are still free to supply non-default values from the environment.)